frontend/src/components/HandbookOverlay/HandbookSystemRole.test.tsx

```diff
@@ -121,4 +124,41 @@ describe('HandbookSystemRole component tests', () => {
 		const lockedBox = container.getElementsByClassName('role-locked');
 		expect(lockedBox).toHaveLength(0);
 	});
+
+	test('renders an error message when no system roles are passed to it', () => {
+		const numLevelsCompleted = 0;
+
+		render(
+			<HandbookSystemRole
+				numCompletedLevels={numLevelsCompleted}
+				systemRoles={emptySystemRoles}
+			/>
+		);
+
+		expect(screen.getByText('System Roles')).toBeInTheDocument();
+
+		// make sure no system roles are displayed on the page
+		expect(screen.queryAllByText(level1SystemRole)).toHaveLength(0);
+		expect(screen.queryAllByText(level2SystemRole)).toHaveLength(0);
+		expect(screen.queryAllByText(level3SystemRole)).toHaveLength(0);
```

Comment:
We don't expect these to occur more than once, plus, we can do better than queryByText because these are role="term":

![image](https://github.com/ScottLogic/prompt-injection/assets/15246391/6970d0e1-7470-4cb7-8003-be0916bc4e56)

So use this instead:

```
expect(screen.queryByRole('term', { name: level1SystemRole })).not.toBeInTheDocument();
```

Reply:
I now see that you copied the existing pattern... so I understand why you did it! But we can improve them all, throughout this file, as suggested.